homeassistant/components/emonitor/sensor.py

```diff
@@ -0,0 +1,108 @@
+"""Support for a Emonitor channel sensor."""
+
+from aioemonitor.monitor import EmonitorChannel
+
+from homeassistant.components.sensor import DEVICE_CLASS_POWER, SensorEntity
+from homeassistant.const import POWER_WATT
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.typing import StateType
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from . import name_short_mac
+from .const import DOMAIN
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up entry."""
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+    channels = coordinator.data.channels
+    entities = []
+    seen_channels = set()
+    for channel_number, channel in channels.items():
+        seen_channels.add(channel_number)
+        if not channel.active:
+            continue
+        if channel.paired_with_channel in seen_channels:
```

Comment:
You can end up adding a channel if the "paired_with_channel" is set up after this channel. Why not check versus `channels` ?

And what if paired with a channel that is not active?

Reply:
Paired with means it uses two CTs to monitor the combined power usage.

Paired with is configured when the system is installed so it's unlikely to change without putting in a new circuit box.

Reply:
Channels are always sequential and the paired channel will always be next to either other.  Ie 1 and 2 can be paired, 3 and 4 can be paired, but not 1 and 7 etc. 

I used a set here since the hash lookup is faster then the list.